### PR TITLE
 fallthrough is now a reserved keyword, use __fallthrough__

### PR DESCRIPTION
**Pull request by harryyoud**
torvalds/linux@294f69e662

### CODE REVIEW

File: include/basic_types.h [hunk 1]
```diff
@@ -14,7 +14,15 @@
...
 #define __BASIC_TYPES_H__
+#include <linux/version.h>
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0))
+#if __has_attribute(__fallthrough__)
+# define fallthrough                    __attribute__((__fallthrough__))
+#else
+# define fallthrough                    do {} while (0)  /* fallthrough */
+#endif
+#endif
```
Comment (harryyoud) [inline include/basic_types.h]: Not super sure where is best for this. Only here because I know it is included just about everywhere else

**Comment by tomaspinho** [comment]
Hey @harryyoud! Is this just for style purposes or is this fixing a compilation error for kernels > 5.4?

**Comment by harryyoud** [comment]
This fixes a compilation error on kernels with the commit I linked. I think this happens because of the original commit turning off the fallthrough attribute on right line 43

**Comment by harryyoud** [comment]
You know what, I think I may have a better solution than this, don't merge this just yet

**Comment by harryyoud** [comment]
This fixes the issue and is much less messy

**Comment by tomaspinho** [comment]
Looks better! Merging

**Pull request merged**